sdk/eventhub/event-hubs/test/client.spec.ts

```diff
@@ -10,304 +10,374 @@ import chaiString from "chai-string";
-  // tslint:disable-next-line: no-null-keyword
-  return Array.apply(null, new Array(length)).map((x: any, i: any) => {
-    return `${i}`;
+  it("creates an EventHubClient from a custom TokenCredential", function(): void {
+    const dummyCredential: TokenCredential = {
+      getToken: async () => {
+        return {
+          token: "boo",
+          expiresOnTimestamp: 12324
+        };
+      }
+    };
+    const client = new EventHubClient("abc", "my-event-hub-path", dummyCredential);
+    client.should.be.an.instanceof(EventHubClient);
+    should.equal(client.eventHubName, "my-event-hub-path");
   });
-}
-
-before("validate environment", function(): void {
-  should.exist(
-    env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
-    "define EVENTHUB_CONNECTION_STRING in your environment before running integration tests."
-  );
-  should.exist(
-    env[EnvVarKeys.EVENTHUB_NAME],
-    "define EVENTHUB_NAME in your environment before running integration tests."
-  );
-});
 
-const service = {
-  connectionString: env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
-  path: env[EnvVarKeys.EVENTHUB_NAME]
-};
+  // it("creates an EventHubClient from an Azure.Identity credential", async function(): Promise<void> {
+  //   should.exist(
+  //     env[EnvVarKeys.AZURE_CLIENT_ID],
+  //     "define AZURE_CLIENT_ID in your environment before running integration tests."
+  //   );
+  //   should.exist(
+  //     env[EnvVarKeys.AZURE_TENANT_ID],
+  //     "define AZURE_TENANT_ID in your environment before running integration tests."
+  //   );
+  //   should.exist(
+  //     env[EnvVarKeys.AZURE_CLIENT_SECRET],
+  //     "define AZURE_CLIENT_SECRET in your environment before running integration tests."
+  //   );
+  //   should.exist(env[EnvVarKeys.ENDPOINT], "define ENDPOINT in your environment before running integration tests.");
+
+  //   const credential = new EnvironmentCredential();
+  //   const client = new EventHubClient(env.ENDPOINT, env.EVENTHUB_NAME, credential);
+
+  //   // Extra check involving actual call to the service to ensure this works
+  //   const hubInfo = await client.getProperties();
+  //   should.equal(hubInfo.path, client.eventHubName);
+  //   await client.close();
+  // });
+});
 
-describe("EventHubClient on #RunnableInBrowser", function(): void {
+describe("ServiceCommunicationError for non existent namespace", function(): void {
   let client: EventHubClient;
 
-  afterEach("close the connection", async function(): Promise<void> {
-    if (client) {
-      debug(">>>>>>>> afterEach: closing the client.");
-      await client.close();
+  beforeEach(() => {
+    client = new EventHubClient("Endpoint=sb://a;SharedAccessKeyName=b;SharedAccessKey=c;EntityPath=d");
+  });
+
+  afterEach(() => {
+    return client.close();
+  });
+
+  it("should throw ServiceCommunicationError while getting hub runtime info", async function(): Promise<void> {
+    try {
+      await client.getProperties();
```

Comment:
For tests that should verify an error is thrown, we should throw an error in the `try` block as well. Otherwise if the operation we're testing doesn't throw an error, the test will still pass since the assertions are all in the `catch block`.

This would be the case for all the tests in this file that assert on an error.

Reply:
For this section of tests, I only moved things around, but your point is right. Can you log an issue to review all tests keeping this lens in mind? I am sure there are quite a few cases like this that we inherited.

One way around this we used in Service Bus (and you will see it in the `errors after close()` section that I added) was to a have a variable to store the error and the test the error message outside of the catch block.

Reply:
Created #3999 to address reviewing tests.